ibis/backends/tests/test_register.py

```diff
@@ -39,6 +39,19 @@ def pushd(new_dir):
         os.chdir(previous_dir)
 
 
+def drop(table: ibis.Table):
+    backend = table._find_backend()
+    name = table.get_name()
+    try:
+        backend.drop_table(name)
```

Comment:
This is one example use case that may inform the API we design in https://github.com/ibis-project/ibis/issues/8382